synapse/http/proxyagent.py

```diff
@@ -237,7 +244,7 @@ def request(self, method, uri, headers=None, bodyProducer=None):
                 self._reactor, parsed_uri.host, parsed_uri.port, **self._endpoint_kwargs
             )
 
-        logger.debug("Requesting %s via %s", uri, endpoint)
+        logger.debug(f"Requesting {uri !s} via {endpoint !s}")
```

Comment:
`logger.debug` (and other logger.*` methods) does its string interpolation lazily: ie, it will not interpolate unless the line is actually going to be logged. It's best to leave this as `%s` etc.